jetty-maven-plugin/src/test/java/org/eclipse/jetty/maven/plugin/MockShutdownMonitor.java

```diff
@@ -0,0 +1,97 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.maven.plugin;
+
+import java.net.ServerSocket;
+
+import org.eclipse.jetty.toolchain.test.IO;
+
+/**
+ * MockShutdownMonitor
+ * 
+ * A helper class that grabs a ServerSocket, spawns a thread and then
+ * passes the ServerSocket to the Runnable. This class has a main so
+ * that it can be used for forking, to mimic the actions of the
+ * org.eclipse.jetty.server.ShutdownMonitor.
+ *
+ */
+public class MockShutdownMonitor
```

Comment:
Why can't we use `ShutdownMonitor`? 
It should be possible because there are tests that use `ShutdownMonitor` directly and not a mock.
We really want to test the real `ShutdownMonitor`, not this mock class.

Reply:
@sbordet I'm not interested in testing ShutdownMonitor - we have other tests for that ;). The unit test it relates to is of the JettyStopMojo - all I want to do is to test the behaviour of the JettyStopMojo, and for that I need to control the behaviour of the ShutdownMonitor and its ShutdownMonitorRunnable. I can't do that by subclassing because ShutdownMonitor is effectively  not extensible, hence the mocks.

Reply:
I made `ShutdownMonitor.start()` and `stop()` public and I can rewrite these tests using directly `ShutdownMonitor` without 200 lines of mock code that must to be bug-free and maintained.
I'd prefer to fix `ShutdownMonitor` than adding a mock for it.

Reply:
@sbordet I have changed the final test `testStopWait` to talk to the real `ShutdownMonitor`. None of the other tests can use the real class because I need to test failure/fallback modes, and `ShutdownMonitor` is unsubclassable (private no-args constructor).